docs/source/library-user-guide/datafusion-query-optimizer.md

```diff
@@ -0,0 +1,336 @@
+<!---
```

Comment:
Perhaps can we name this file `query-optimizer.md` rather than `datafusion-query-optimizer.md`? I think the datafusion is redundant